### PR TITLE
Add GitHub integration to ProfilePage (#230)

### PR DESCRIPTION
**Pull request by yadavshubham01**
Description :-
I have implemented GitHub integration within the Profile Page component of our application. The following steps outline the process I followed to create this integration:

Feature Implementation:-

Integrated GitHub OAuth functionality to allow users to connect their GitHub accounts from their profile settings.
Modified the ProfilePage component to include a button for connecting GitHub, displaying the integration status, and providing visual feedback on the connection state.
Utilized the Octokit library for making API calls to GitHub and managing user installations effectively.

Changes Made:-
Updated github.service.js to handle authentication and data fetching for connected GitHub accounts.
Enhanced the github.controller.js to manage the GitHub callback and webhook processing for issues and pull requests.
Adjusted the route definitions in github.route.js to include the new integration endpoint and callback handler.
Made UI improvements to the ProfilePage component to display integration status and manage user interactions more effectively.

### CODE REVIEW

**Comment by yadavshubham01** [comment]
@oliursahin 
please review this pr

**Comment by thestoneroller** [comment]
@yadavshubham01 In your code, I noticed that the Twilio-related services are commented out. Is that because of the errors while setting up the dev environment? If so, you can use placeholder values for the environment variables in the backend `.env` file.

Here’s what you can do:

```bash
TWILIO_ACCOUNT_SID=ACCOUNTSID # Must start with 'AC'
TWILIO_AUTH_TOKEN=authToken
```

**Comment by sajdakabir** [comment]
@yadavshubham01 , Please resolve the conflicts.

**Comment by yadavshubham01** [comment]
Wait I will correct it

**Comment by yadavshubham01** [comment]
@sajdakabir @TanmayAdithya done 👍

File: apps/frontend/src/app/(app)/profile/page.tsx [hunk 1]
```diff
@@ -5,6 +5,22 @@ import { IntegrationList } from "@/src/components/atoms/IntegrationList"
...
+        const data = await response.json()
+        window.location.href = data.url 
+      } else {
+        console.error("Failed to initiate GitHub integration")
+      }
+    } catch (error) {
+      console.error("Error connecting to GitHub:", error)
+    }
+  }
+
```
Comment (madhav-relish) [inline apps/frontend/src/app/(app)/profile/page.tsx]: Follow the same code structure as used in today and inbox page.

File: apps/frontend/src/app/(app)/profile/page.tsx [hunk 2]
```diff
@@ -42,6 +58,19 @@ const ProfilePage: React.FC = () => {
...
+              <p className="text-green-400">GitHub account connected successfully!</p>
+            ) : (
+              <button
+                onClick={handleConnectGithub}
+                className="mt-2 bg-blue-600 text-white py-2 px-4 rounded"
+              >
+                Connect GitHub
+              </button>
+            )}
+          </div>
```
Comment (madhav-relish) [inline apps/frontend/src/app/(app)/profile/page.tsx]: I don't think any of this is required! We already have the buttons in place

image

File: apps/frontend/src/app/(app)/profile/page.tsx [hunk 3]
```diff
@@ -5,6 +5,22 @@ import { IntegrationList } from "@/src/components/atoms/IntegrationList"
 import { ProfilePicture } from "@/src/components/atoms/ProfilePicture"
 
 const ProfilePage: React.FC = () => {
+  const [isGithubConnected, setIsGithubConnected] = React.useState(false)
+
+  const handleConnectGithub = async () => {
+    try {
+      const response = await fetch("/integrate/github", { method: "POST" })
```
Comment (madhav-relish) [inline apps/frontend/src/app/(app)/profile/page.tsx]: This path doesn't exist.

**Review by madhav-relish — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/(app)/profile/page.tsx (hunk 2 above)
Comment (immadsahin) [reply apps/frontend/src/app/(app)/profile/page.tsx]: which app it is @madhav-relish 😿

**Comment by yadavshubham01** [comment]
hey @madhav-relish  review it i made some change

File: apps/backend/package.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
     "crypto-random-string": "^5.0.0",
     "dotenv": "^16.4.5",
     "express": "^4.19.2",
+    "express-session": "^1.18.1",
```
Comment (madhav-relish) [inline apps/backend/package.json]: This doesn't seem to be used anywhere in the app.

**Review by madhav-relish — CHANGES_REQUESTED** [review]
Update the .env.example files(both FE and BE in this case), and add the steps to setup.
And resolve the merge conflicts

File: apps/backend/package.json (hunk 4 above)
Comment (yadavshubham01) [reply apps/backend/package.json]: yes ,I will remove it

**Comment by sajdakabir** [comment]
Hey @yadavshubham01 , your branch is having a lot of conflicts. Can you please fix them so that we can move forward with this PR?

**Comment by yadavshubham01** [comment]
hey @madhav-relish @TanmayAdithya  i just update my PR please review it

**Comment by sajdakabir** [comment]
Hey @yadavshubham01 , it still has conflicts. Can you please fix them?
Screenshot 2024-10-14 at 7 14 18 PM

**Pull request closed without merging**